homeassistant/components/vicare/sensor.py

```diff
@@ -0,0 +1,289 @@
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    TEMP_CELSIUS,
+    UNIT_PERCENTAGE,
+)
+from homeassistant.helpers.entity import Entity
+
+from . import (
+    DOMAIN as VICARE_DOMAIN,
+    PYVICARE_ERROR,
+    VICARE_API,
+    VICARE_HEATING_TYPE,
+    VICARE_NAME,
+    HeatingType,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GETTER = "getter"
+
+SENSOR_TYPE_TEMPERATURE = "temperature"
+
+SENSOR_OUTSIDE_TEMPERATURE = "outside_temperature"
+SENSOR_SUPPLY_TEMPERATURE = "supply_temperature"
+SENSOR_RETURN_TEMPERATURE = "return_temperature"
+
+# gas sensors
+SENSOR_BOILER_TEMPERATURE = "boiler_temperature"
+SENSOR_BURNER_MODULATION = "burner_modulation"
+SENSOR_BURNER_STARTS = "burner_starts"
+SENSOR_BURNER_HOURS = "burner_hours"
+SENSOR_BURNER_POWER = "burner_power"
+SENSOR_DHW_GAS_CONSUMPTION_TODAY = "hotwater_gas_consumption_today"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_WEEK = "hotwater_gas_consumption_heating_this_week"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_MONTH = "hotwater_gas_consumption_heating_this_month"
+SENSOR_DHW_GAS_CONSUMPTION_THIS_YEAR = "hotwater_gas_consumption_heating_this_year"
+SENSOR_GAS_CONSUMPTION_TODAY = "gas_consumption_heating_today"
+SENSOR_GAS_CONSUMPTION_THIS_WEEK = "gas_consumption_heating_this_week"
+SENSOR_GAS_CONSUMPTION_THIS_MONTH = "gas_consumption_heating_this_month"
+SENSOR_GAS_CONSUMPTION_THIS_YEAR = "gas_consumption_heating_this_year"
+
+# heatpump sensors
+SENSOR_COMPRESSOR_STARTS = "compressor_starts"
+SENSOR_COMPRESSOR_HOURS = "compressor_hours"
+
+SENSOR_TYPES = {
+    SENSOR_OUTSIDE_TEMPERATURE: {
+        CONF_NAME: "Outside Temperature",
+        CONF_ICON: None,
+        CONF_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+        CONF_GETTER: lambda api: api.getOutsideTemperature(),
+        CONF_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+    },
+    SENSOR_SUPPLY_TEMPERATURE: {
+        CONF_NAME: "Supply Temperature",
+        CONF_ICON: None,
+        CONF_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+        CONF_GETTER: lambda api: api.getSupplyTemperature(),
+        CONF_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+    },
+    # gas sensors
+    SENSOR_BOILER_TEMPERATURE: {
+        CONF_NAME: "Boiler Temperature",
+        CONF_ICON: None,
+        CONF_UNIT_OF_MEASUREMENT: TEMP_CELSIUS,
+        CONF_GETTER: lambda api: api.getBoilerTemperature(),
+        CONF_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
+    },
+    SENSOR_BURNER_MODULATION: {
+        CONF_NAME: "Burner modulation",
+        CONF_ICON: "mdi:percent",
+        CONF_UNIT_OF_MEASUREMENT: UNIT_PERCENTAGE,
+        CONF_GETTER: lambda api: api.getBurnerModulation(),
+        CONF_DEVICE_CLASS: None,
+    },
+    SENSOR_DHW_GAS_CONSUMPTION_TODAY: {
+        CONF_NAME: "Hot water gas consumption today",
+        CONF_ICON: None,
+        CONF_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+        CONF_GETTER: lambda api: api.getGasConsumptionDomesticHotWaterToday(),
+        CONF_DEVICE_CLASS: DEVICE_CLASS_POWER,
```

Comment:
Energy measurements with unit kWh should not have device class power.

Reply:
Ok thanks, I've changed it back :)